concordium-consensus/src/Concordium/KonsensusV1/Consensus/CatchUp.hs

```diff
@@ -435,8 +435,8 @@ processCatchUpTerminalData ::
     CatchUpTerminalData ->
     m TerminalDataResult
 processCatchUpTerminalData CatchUpTerminalData{..} = flip runContT return $ do
-    progress0 <- processFE False cutdLatestFinalizationEntry
-    progress1 <- foldM processQC progress0 cutdHighestQuorumCertificate
+    progress0 <- foldM processQC False cutdHighestQuorumCertificate
+    progress1 <- processFE progress0 cutdLatestFinalizationEntry
```

Comment:
Seeing that there seemed to have been a bug here, can you please add documentation as to why this order is correct, and not the previous one?

Reply:
The old ordering isn't wrong per se, but this is more efficient. The bug was caused by relying on `roundExistingQCs` to determine if the QC was worth processing. While the finalization entry was generally the trigger for the bug (because it was processed first), the bug could still potentially be triggered even with the reordering.

Reply:
Thanks. But if this is more efficient than that is still useful insight that should be documented.